contracts/crowdsale/validation/TimedCrowdsale.sol

```diff
@@ -0,0 +1,55 @@
+pragma solidity ^0.4.18;
+
+import "../../math/SafeMath.sol";
+import "../Crowdsale.sol";
+
+
+/**
+ * @title TimedCrowdsale
+ * @dev Crowdsale accepting contributions only within a time frame.
+ */
+contract TimedCrowdsale is Crowdsale {
+  using SafeMath for uint256;
+
+  uint256 public startTime;
+  uint256 public endTime;
+
+  /**
+   * @dev Reverts if not in crowdsale time range. 
+   */
+  modifier inTimeRange {
+    require(now >= startTime && now <= endTime);
+    _;
+  }
+
+  /**
+   * @dev Constructor, takes crowdsale opening and closing times.
+   * @param _startTime Crowdsale opening time
+   * @param _endTime Crowdsale closing time
+   */
+  function TimedCrowdsale(uint256 _startTime, uint256 _endTime) public {
+    require(_startTime >= now);
+    require(_endTime >= _startTime);
+
+    startTime = _startTime;
+    endTime = _endTime;
+  }
+
+  /**
+   * @dev Checks whether the period in which the crowdsale is open has already elapsed.
+   * @return Whether crowdsale period has elapsed
+   */
+  function hasExpired() public view returns (bool) {
```

Comment:
any reason we're not using the existing naming of `hasEnded`? I don't mind either way, but it's good to know why we changed the api.

Reply:
`hasEnded` was the prior method that encompassed ending either by reaching `endTime` or by reaching the cap in a `CappedCrowdsale`. Now the purchase validation is done in separate places so we need to split this (`hasExpired` and `capReached`). While we could call this one `hasEnded`, a crowdsale could also end "prematurely" by reaching its cap, thus, be ended but not expired..